From a5100d0798e61f90d795262220d6acc353185ee9 Mon Sep 17 00:00:00 2001
From: Luo Meng <luomeng12@huawei.com>
Date: Mon, 21 Nov 2022 22:03:09 +0800
Subject: [PATCH] dm ioctl: add DMINFO() to track dm device create/remove
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: a5100d0798e61f90d795262220d6acc353185ee9
Modified-by-SEL: No


hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I61HSS
CVE: NA

--------------------------------

Add DMINFO() to help tracking device creation/removal success.

Signed-off-by: Luo Meng <luomeng12@huawei.com>
Reviewed-by: Zhang Yi <yi.zhang@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/md/dm-ioctl.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/md/dm-ioctl.c b/drivers/md/dm-ioctl.c
index b012a2748af8..2186a3a4e48b 100644
--- a/drivers/md/dm-ioctl.c
+++ b/drivers/md/dm-ioctl.c
@@ -272,6 +272,9 @@ static struct dm_table *__hash_remove(struct hash_cell *hc)
 	table = NULL;
 	if (hc->new_map)
 		table = hc->new_map;
+
+	DMINFO("%s[%i]: %s (%s) is removed successfully",
+		current->comm, current->pid, hc->md->disk->disk_name, hc->name);
 	dm_put(hc->md);
 	free_cell(hc);
 
@@ -773,6 +776,7 @@ static int dev_create(struct file *filp, struct dm_ioctl *param, size_t param_si
 {
 	int r, m = DM_ANY_MINOR;
 	struct mapped_device *md;
+	struct hash_cell *hc;
 
 	r = check_name(param->name);
 	if (r)
@@ -796,6 +800,9 @@ static int dev_create(struct file *filp, struct dm_ioctl *param, size_t param_si
 
 	__dev_status(md, param);
 
+	hc = dm_get_mdptr(md);
+	DMINFO("%s[%i]: %s (%s) is created successfully",
+		current->comm, current->pid, md->disk->disk_name, hc->name);
 	dm_put(md);
 
 	return 0;
-- 
2.33.0

